README.md

```diff
@@ -82,6 +82,74 @@ model = SetFitModel.from_pretrained("lewtun/my-awesome-setfit-model")
 preds = model(["i loved the spiderman movie!", "pineapple on pizza is the worst 🤮"]) 
 ```
 
+Here is an end-to-end example using `SetFitHead`:
+
+
+```python
+from datasets import load_dataset
+from sentence_transformers.losses import CosineSimilarityLoss
+
+from setfit import SetFitModel, SetFitTrainer
+
+
+# Load a dataset from the Hugging Face Hub
+dataset = load_dataset("sst2")
+
+# Simulate the few-shot regime by sampling 8 examples per class
+num_classes = 2
+train_dataset = dataset["train"].shuffle(seed=42).select(range(8 * num_classes))
+eval_dataset = dataset["validation"]
+
+# Load a SetFit model from Hub
+model = SetFitModel.from_pretrained(
+    "sentence-transformers/paraphrase-mpnet-base-v2",
+    use_differentiable_head=True,
+    head_params={"out_features": num_classes},
```

Comment:
Unrelated to this PR, but I've been wondering if it would be clearer to just have `num_labels` as an arg instead of `head_params`. 

Are there some use cases where it makes sense to expose the full set of `head_params`?

Reply:
I remember I wanted to align the same APIs as the [sklearn's head](https://github.com/huggingface/setfit/blob/main/tests/test_modeling.py#L56-L71).

I guess the idea was to separate the head's parameters from others? But ya, it's not pretty intuitive.

Reply:
Thanks, that makes sense. I'll play around and might consider a refactor in future :)